### PR TITLE
Initial Integration test minimal default values

### PR DESCRIPTION
**Pull request by flexyford**
Setup an integration test with the default constructor values. And setup the fetch function in parent beforeEach for the entire test suite.

Copied the test-server from `impagination` and return
`Ember.RSVP.Promises` instead of `Promises`

Remove ember-impagination unit tests because the `impagination` npm
package already performs the same tests.

### CODE REVIEW

File: tests/integration/components/impagination-collection-test.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/*global it, beforeEach */
```
Comment (Robdel12) [inline tests/integration/components/impagination-collection-test.js]: You could import these :P

Comment (flexyford) [inline tests/integration/components/impagination-collection-test.js]: From where? Ember-Mocha!

Comment (Robdel12) [inline tests/integration/components/impagination-collection-test.js]: Yeah line 6 you should be able to import it

File: tests/test-helper.js [hunk 2]
```diff
@@ -2,3 +2,30 @@ import resolver from './helpers/resolver';
 import { setResolver } from 'ember-mocha';
 
 setResolver(resolver);
+
+// Polyfill: Array.prototype.fill()
+if (!Array.prototype.fill) {
```
Comment (cowboyd) [inline tests/test-helper.js]: This shouldn't be necessary with Phantom2

File: tests/test-server.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
```
Comment (cowboyd) [inline tests/test-server.js]: Is this file necessary?

Comment (flexyford) [inline tests/test-server.js]: It would be a simple way to view the Pending, Resolved, and Rejected effects on the `records` and  `datasetState`. It could be considered overkill, but I'm imagining the whole test suite again as
- Requesting Records
  - Resolving Records
  - Rejecting Records

File: tests/test-helper.js (hunk 2 above)
Comment (Robdel12) [inline tests/test-helper.js]: I think this is because prototype extensions in addons are disabled.
